sdk/core/azure-core/azure/core/paging_method.py

```diff
@@ -0,0 +1,225 @@
+        response = self._client._pipeline.run(request, stream=False)  # pylint: disable=protected-access
+
+        http_response = response.http_response
+        if not 200 <= http_response.status_code < 300:
+            map_error(status_code=http_response.status_code, response=http_response, error_map=self._error_map)
+            raise HttpResponseError(response=http_response)
+
+        return response
+
+    def finished(self, continuation_token):
+        # type: (Any) -> bool
+        return continuation_token is None and self.did_a_call_already
+
+    def get_list_elements(self, pipeline_response, deserialized):
+        # type: (HttpResponse, ResponseType) -> Iterable[ReturnType]
+        if not hasattr(deserialized, self._item_name):
+            raise ValueError(
+                "The response object does not have property '{}' to extract element list from".format(self._item_name)
+            )
+        return getattr(deserialized, self._item_name)
+
+    def mutate_list(self, pipeline_response, list_of_elem):
+        # type: (HttpResponse, Iterable[ReturnType]) -> Iterable[ReturnType]
+        if self._cls:
+            list_of_elem = self._cls(list_of_elem)
+        return iter(list_of_elem)
+
+    def get_continuation_token(self, pipeline_response, deserialized):
+        # type: (HttpResponse, ResponseType) -> Any
+        if not self._next_link_name:
+            return None
+        if not hasattr(deserialized, self._next_link_name):
+            raise ValueError(
+                "The response object does not have property '{}' to extract continuation token from".format(
+                    self._next_link_name
+                )
+            )
+        return getattr(deserialized, self._next_link_name)
+
+    def extract_data(self, pipeline_response):
+        # type: (HttpResponse) -> Tuple[Any, Iterable[ReturnType]]
+        deserialized = self._deserialize_output(pipeline_response)
+        list_of_elem = self.get_list_elements(pipeline_response, deserialized)  # type: Iterable[ReturnType]
+        list_of_elem = self.mutate_list(pipeline_response, list_of_elem)
+        continuation_token = self.get_continuation_token(pipeline_response, deserialized)
+        return continuation_token, list_of_elem
+
+
+class DifferentNextOperationPagingMethod(BasicPagingMethod):
+    """Use this paging method if the swagger defines a different next operation
+    """
+
+    def __init__(self):
+        super(DifferentNextOperationPagingMethod, self).__init__()
+        self._prepare_next_request = None
+
+    def initialize(self, client, deserialize_output, next_link_name, **kwargs):
+        # type: (PipelineClient, Callable, str, Any) -> None
+        super(DifferentNextOperationPagingMethod, self).initialize(
+            client, deserialize_output, next_link_name, **kwargs
+        )
+        self._prepare_next_request = kwargs.pop("prepare_next_request", None)
```

Comment:
```python
try:
    self._prepare_next_request = kwargs.pop("prepare_next_request")
except KeyError:
    raise TypeError("<insert prefix here> is missing required keyword-only argument 'prepare_next_request'"
```

Reply:
(in other words, TypeError for *missing* parameter, ValueError for present, but wrong value...)